### PR TITLE
Feature flags groups support & /decide refactor

### PR DESCRIPTION
**Pull request by macobo**
## Changes

This PR adds support for groups in feature flags in posthog-js.

In theory it could have been a ~3 line change, but this would result in a _lot_ of extra and racy /decide calls.

To have it working sensibly, I also need to cut down on /decide requests:
- we wait until `.loaded()` has been called before calling decide for the first time (to make sure we have all the parameters)
- We batch `.reloadFeatureFlags` calls to only happen once

The batching logic is tested in cypress tests to verify intended behavior.

## Checklist
- [x] Tests for new code (if applicable)
- [x] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

File: src/posthog-core.js [hunk 1]
```diff
@@ -218,11 +212,7 @@ PostHogLib.prototype.init = function (token, config, name) {
         return
     }
 
-    var instance = create_mplib(token, config, name)
-    posthog_master[name] = instance
```
Comment (hazzadous) [inline src/posthog-core.js:222]: readd this @macobo

Comment (hazzadous) [reply src/posthog-core.js:222]: Is there a test that would have failed?

Comment (macobo) [reply src/posthog-core.js:222]: Sadly nope - the library init code is full of gremlins.

File: cypress/integration/capture.spec.js [hunk 2]
```diff
@@ -97,6 +96,20 @@ describe('Event capture', () => {
         cy.phCaptures().should('not.include', '$rageclick')
     })
 
+    it('makes a single decide request', () => {
+        start()
+
+        cy.wait(200)
+        cy.shouldBeCalled('decide', 1)
```
Comment (EDsCODE) [inline cypress/integration/capture.spec.js:103]: 👍

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**